lib/src/RedisRemoteSaiInterface.cpp

```diff
@@ -827,6 +827,106 @@ sai_status_t RedisRemoteSaiInterface::waitForObjectTypeGetAvailabilityResponse(
     return status;
 }
 
+sai_status_t RedisRemoteSaiInterface::queryAattributeCapability(
+        _In_ sai_object_id_t switchId,
+        _In_ sai_object_type_t objectType,
+        _In_ sai_attr_id_t attrId,
+        _Out_ sai_attr_capability_t *capability)
+{
+    SWSS_LOG_ENTER();
+
+    auto switch_id_str = sai_serialize_object_id(switchId);
+    auto object_type_str = sai_serialize_object_type(objectType);
+
+    auto meta = sai_metadata_get_attr_metadata(objectType, attrId);
+
+    if (meta == NULL)
+    {
+        SWSS_LOG_ERROR("Failed to find attribute metadata: object type %s, attr id %d", object_type_str.c_str(), attrId);
+        return SAI_STATUS_INVALID_PARAMETER;
+    }
+
+    const std::string attr_id_str = meta->attridname;
+
+    const std::vector<swss::FieldValueTuple> entry =
+    {
+        swss::FieldValueTuple("OBJECT_TYPE", object_type_str),
+        swss::FieldValueTuple("ATTR_ID", attr_id_str)
+    };
+
+    SWSS_LOG_DEBUG(
+            "Query arguments: switch %s, object type: %s, attribute: %s",
+            switch_id_str.c_str(),
+            object_type_str.c_str(),
+            attr_id_str.c_str()
+    );
+
+    // This query will not put any data into the ASIC view, just into the
+    // message queue
+
+    m_recorder->recordQueryAattributeCapability(switchId, objectType, attrId, capability);
+
+    m_redisChannel->set(switch_id_str, entry, REDIS_ASIC_STATE_COMMAND_ATTR_CAPABILITY_QUERY);
+
+    auto status = waitForQueryAattributeCapabilityResponse(capability);
+
+    m_recorder->recordQueryAattributeCapabilityResponse(status, objectType, attrId, capability);
+
+    return status;
+}
+
+sai_status_t RedisRemoteSaiInterface::waitForQueryAattributeCapabilityResponse(
+        _Out_ sai_attr_capability_t* capability)
+{
+    SWSS_LOG_ENTER();
+
+    swss::KeyOpFieldsValuesTuple kco;
+
+    auto status = m_redisChannel->wait(REDIS_ASIC_STATE_COMMAND_ATTR_CAPABILITY_RESPONSE, kco);
+
+    if (status == SAI_STATUS_SUCCESS)
+    {
+        const std::vector<swss::FieldValueTuple> &values = kfvFieldsValues(kco);
+
+        if (values.size() != 3)
+        {
+            SWSS_LOG_ERROR("Invalid response from syncd: expected 3 values, received %zu", values.size());
+
+            return SAI_STATUS_FAILURE;
+        }
+
+        const uint32_t create_implemented = std::stoi(fvValue(values[0]));
```

Comment:
those are numbers ? instead of string "true"/"false" ?

Reply:
yes they are number strings "0 or 1" instead of "true/false"
See syncd/Syncd.cpp:
entry =
        {
            swss::FieldValueTuple("CREATE_IMPLEMENTED", std::to_string(capability.create_implemented)),
            swss::FieldValueTuple("SET_IMPLEMENTED", std::to_string(capability.set_implemented)),
            swss::FieldValueTuple("GET_IMPLEMENTED", std::to_string(capability.get_implemented))
        };

Reply:
could you make them as true false strings? all other bool that are serialized are strings like that

Reply:
Agreed. I have made the change to not use "number string" to follow the common practice to use "true/false string".
In order to make this change I have also changed syncd/Syncd.cpp line 355-361 to write with "true/false string".
Please help review that portion of code as well in addition to this change here.